src/processor/operator/persistent/rel_batch_insert.cpp

```diff
@@ -225,5 +226,15 @@ void RelBatchInsert::finalizeInternal(ExecutionContext* context) {
     partitionerSharedState->partitioningBuffers[relInfo->partitioningIdx].reset();
 }
 
+void RelBatchInsert::updateProgress(ExecutionContext* context) {
+    std::lock_guard<std::mutex> lock(progressSharedState->mtx);
+    progressSharedState->partitionsDone = partitionerSharedState->nextPartitionIdx;
+    progressSharedState->partitionsTotal =
+        partitionerSharedState->numPartitions[partitionerSharedState->numPartitions.size() - 1];
+    double progress =
+        double(progressSharedState->partitionsDone) / double(progressSharedState->partitionsTotal);
+    context->clientContext->getProgressBar()->updateProgress(context->queryID, progress);
+}
```

Comment:
This can be simplified.

```
void RelBatchInsert::updateProgress(ExecutionContext* context) {
    double progress =
        double(progressSharedState->partitionsDone) / double(progressSharedState->partitionsTotal);
    context->clientContext->getProgressBar()->updateProgress(context->queryID, progress);
}
```